src/ConnectSdk/Querying/Query.cs

```diff
@@ -50,21 +55,53 @@ public Query(string collection, IEventEndpoint endpoint, AliasedAggregations agg
             GroupBy = groups;
             Interval = interval;
             Timezone = timezone;
-        }
 
-        public virtual Query<TNewResultType> UpdateWith<TNewResultType>(AliasedAggregations aggregations = null, FilteredProperties filters = null, ITimeframe timeframe = null, string[] groups = null, Interval? interval = null, object timezone = null)
+            if (custom == null)
+                return;
+
+            _data = new Dictionary<string, object>();
+
+            foreach (var pair in custom)
+            {
+                _data.Add(pair.Key, pair.Value);
+            }
+        }
+        
+        public virtual Query<TNewResultType> UpdateWith<TNewResultType>(AliasedAggregations aggregations = null, FilteredProperties filters = null, ITimeframe timeframe = null, string[] groups = null, Interval? interval = null, object timezone = null, IEnumerable<KeyValuePair<string, object>> custom = null)
         {
-            return new  Query<TNewResultType>(Collection, Endpoint, aggregations ?? this.Select, filters ?? this.Filter, timeframe ?? this.Timeframe, groups ?? this.GroupBy, interval ?? this.Interval, timezone ?? this.Timezone);
+            return new Query<TNewResultType>(Collection, Endpoint, aggregations ?? this.Select, filters ?? this.Filter, timeframe ?? this.Timeframe, groups ?? this.GroupBy, interval ?? this.Interval, timezone ?? this.Timezone, custom ?? this.Custom);
         }
 
         public virtual async Task<QueryResponse<TResult>> Execute()
         {
              return await Endpoint.Query(Collection, this);
         }
-        
+
+        private void OverrideReservedValue(Dictionary<string, object> dict, string key, object value)
+        {
+            if (value == null)
+                return;
+
+            dict.Add(key, value);
+        }
+
         public override string ToString()
         {
-            return JsonConvert.SerializeObject(this, Formatting.None, Serializer);
+            if (_data == null)
+                return JsonConvert.SerializeObject(this, Formatting.None, Serializer);
+
+            // Create a new temporary dictionary so we don't actually put our other values into the CustomValue dictionary
+            var tempDictionary = _data.ToDictionary(pair => pair.Key, pair => pair.Value);
```

Comment:
We can create an ExpandoObject (or a dynamic object) and apply each of the key value pairs to it, then apply our properties over the top and have that converted to a string. This does still have the issue that if we did want to add more properties to the Query class we have to remember to add them to this object in the ToString.

The issue comes down to how the JsonConverter handles our object. If we want each of the key value pairs to be appended to the Query object as properties we cannot do that dynamically as it's a typed class. 

If we can merge two JSON objects taking the properties from a second one we could convert the dictionary to a JSON object, then the apply the Query object over the top and return the result. That way we wouldn't have to keep track of individual properties, and we could drop the ReservedQueryKeys class, I do like that class though as we can then throw exceptions based on reserved key words. Preventing the user from trying to go `.Custom("select", {})` etc